hypothesis-python/src/hypothesis/extra/array_api.py

```diff
@@ -269,23 +295,46 @@ def check_valid_minmax(prefix, val, info_obj):
             kw["exclude_max"] = exclude_max
 
         return st.floats(width=finfo.bits, **kw)
+    else:
+        # A less-inelegant solution to support complex dtypes exists, but as
+        # this is currently a draft feature, we might as well wait for
+        # discussion of complex inspection to resolve first - a better method
+        # might become available soon enough.
+        # See https://github.com/data-apis/array-api/issues/433
+        for attr in ["float32", "float64", "complex64"]:
+            if not hasattr(xp, attr):
+                raise NotImplementedError(
+                    f"Array module {xp.__name__} has no dtype {attr}, which is "
+                    "currently required for xps.from_dtype() to work with "
+                    "any complex dtype."
+                )
+        kw = {
+            "allow_nan": allow_nan,
+            "allow_infinity": allow_infinity,
+            "allow_subnormal": allow_subnormal,
+        }
+        if dtype == xp.complex64:
+            floats = _from_dtype(xp, api_version, xp.float32, **kw)
+        else:
+            floats = _from_dtype(xp, api_version, xp.float64, **kw)
+
+        return st.builds(complex, floats, floats)
 
 
 class ArrayStrategy(st.SearchStrategy):
-    def __init__(self, xp, elements_strategy, dtype, shape, fill, unique):
+    def __init__(self, xp, api_version, elements_strategy, dtype, shape, fill, unique):
```

Comment:
There are enough argument here that I'd be more comfortable if they were keyword-only after `self`.